packages/kbn-docs-utils/src/api_docs/build_api_declarations/get_type_kind.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Type, Node } from 'ts-morph';
+import { TypeKind } from '../types';
+
+export function getTypeKind(node: Node): TypeKind {
+  if (Node.isTypeAliasDeclaration(node)) {
+    return TypeKind.TypeKind;
+  } else {
+    return getTypeKindForType(node.getType());
+  }
+}
+
+function getTypeKindForType(type: Type): TypeKind {
+  // I think a string literal is also a string... but just in case, checking both.
+  if (type.isString() || type.isStringLiteral()) {
+    return TypeKind.StringKind;
+  } else if (type.isNumber() || type.isNumberLiteral()) {
+    return TypeKind.NumberKind;
+
+    // I could be wrong about this logic. Does this existance of a call signature mean it's a function?
+  } else if (type.getCallSignatures().length > 0) {
+    return TypeKind.FunctionKind;
+  } else if (type.isArray()) {
+    // Arrays are also objects, check this first.
+    return TypeKind.ArrayKind;
+  } else if (type.isObject()) {
+    return TypeKind.ObjectKind;
+  } else if (type.isBoolean()) {
```

Comment:
Do we need this also?

```suggestion
  } else if (type.isBoolean() || type.isBooleanLiteral()) {
```